### PR TITLE
fix: [M3-7105] - Highlight.js Theme Switching

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Fixes bug where Highlight.js theme would not switch
  - The the would not switch because both css files were loaded at the same time. The solution involves switching out the css when the theme is changed.

## Major Changes 🔄
- Made `LinodeThemeWrapper` a named export ✏️
- Moves Highlight.js logic from the `LinodeThemeWrapper` to the `HighlightedMarkdown` so that the css is only imported when `HighlightedMarkdown` is imported. (This helps us seperate concerns and defer loading until it is really needed)
- Removed unnecessary uses of `LinodeThemeWrapper` in some unit tests 🚫
- Removed weird `setTimeout` logic that did nothing 🚫

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2023-09-20 at 3 30 24 PM | Screenshot 2023-09-20 at 3 30 29 PM |

## How to test 🧪

> [!important]
> Please test locally **and** using the internal preview link! We need to verify this fix works on both **dev** and **production** builds.

- Check out this PR
- Spin up your dev server with `yarn dev`
- Go to a Kubernetes Cluster details page and open the Kubeconfig drawer
- Verify that toggling the theme works
- Verify that, upon refreshing the page, the correct theme is rendered
- Repeat similar steps above with the internal preview link to verify that this fix also works for production builds of the app

### CODE REVIEW

File: packages/manager/package.json [hunk 1]
```diff
@@ -271,6 +271,7 @@
       "^react-native$": "react-native-web",
       "ramda": "ramda/src/index.js",
       "^src/(.*)": "<rootDir>/src/$1",
+      "(.*)\\.css\\?raw$": "$1.css",
```
Comment (bnussman-akamai) [inline packages/manager/package.json:274]: Vite supports importing `highlight.js/styles/a11y-dark.css?raw` to get the CSS as a raw string but jest does not by default. This allows our unit tests to understand `.css?raw` imports.

If we were to switch to Vitest, we'd likely wouldn't need this.

File: packages/manager/src/LinodeThemeWrapper.tsx [hunk 2]
```diff
@@ -1,96 +1,30 @@
...
+        {children}
+      </ThemeProvider>
     </StyledEngineProvider>
   );
 };
-
-const toggleTheme = (value: ThemeChoice) => {
-  setTimeout(() => {
-    document.body.classList.remove('no-transition');
-  }, 500);
```
Comment (bnussman-akamai) [inline packages/manager/src/LinodeThemeWrapper.tsx:92]: This does nothing meaningful so I'm removing it. It should also help us on the switch to Vitest if we follow through with that. (It was causing issues if I recall correctly @jdamore-linode)

File: packages/manager/src/env.d.ts [hunk 3]
```diff
@@ -40,3 +40,8 @@ declare module '*.svg' {
   const src: ComponentClass<any, any>;
   export default src;
 }
+
+declare module '*.css?raw' {
+  const src: string;
+  export default src;
+}
```
Comment (bnussman-akamai) [inline packages/manager/src/env.d.ts:47]: This enables typescript support for `*.css?raw` imports. This is similar to how we allow Typescript to allow `.svg` imports

**Review by jaalah-akamai — APPROVED** [review]
- Confirmed that markdown is rendering properly in both light/dark mode and that the theme remains unchanged on refresh
- Confirmed the same with the internal preview link

**Pull request merged**